### PR TITLE
examples: fix (2022-12)

### PR DESCRIPTION
**Pull request by maiha**
Happy Christmas to Crystal!
Let's catch up examples before the New Year. 😃 

* checker: https://github.com/maiha/crystal-examples (1.3.0)
* compiler: Crystal 1.7.0-dev [63e46ebdf] (2022-12-23)
* last commit: 63e46ebdf

This version newly supports the following comment literals.
```crystal
x # => 10.00:00:00.000010000
x # => UUID(ba714f86-cac6-42c7-8956-bcf5105e1b81)
x # => 2016-02-15 04:35:50.0 UTC
x # => NaN
x # => Foo(@a=1, @b=1.0)
```

This increased coverage from 75.3% to 85.8%.

```
75.3%(1489/1979)  1489 success 44 pending 3 general 142 pseudo 70 macro 9 wrong 44 random
↓
85.8%(1725/2010)  1725 success 4 pending 5 general 103 pseudo 72 macro 14 wrong 87 random
```

Best regards,

### CODE REVIEW

File: src/time.cr [hunk 1]
```diff
@@ -84,7 +84,7 @@ require "crystal/system/time"
 #
 # ```
 # time = Time.local(2018, 3, 8, 22, 5, 13, location: Time::Location.load("Europe/Berlin"))
-# time          # => 2018-03-08 22:05:13 +01:00 Europe/Berlin
+# time.to_s     # => "2018-03-08 22:05:13 +01:00"
```
Comment (straight-shoota) [inline src/time.cr:87]: This change looks odd. In the next example there's also just `time # =>` with its inspect value.

File: src/hash.cr [hunk 2]
```diff
@@ -1102,8 +1102,8 @@ class Hash(K, V)
   #
   # ```
   # h = {"a" => {"b" => [10, 20, 30]}}
-  # h.dig? "a", "b"                # => [10, 20, 30]
-  # h.dig? "a", "b", "c", "d", "e" # => nil
+  # h.dig? "a", "b" # => [10, 20, 30]
+  # h.dig? "x", "a" # => nil
```
Comment (Sija) [inline src/hash.cr:1106]: Let's keep consistency between the examples (see `dig`)

```suggestion
  # h.dig? "a", "x" # => nil
```

File: src/named_tuple.cr [hunk 3]
```diff
@@ -243,9 +243,9 @@ struct NamedTuple
   # Returns `nil` if not found.
   #
   # ```
-  # h = {a: {b: [10, 20, 30]}}
-  # h.dig? "a", "b"                # => [10, 20, 30]
-  # h.dig? "a", "b", "c", "d", "e" # => nil
+  # h = {a: {b: {c: [10, 20]}}, x: {a: "b"}}
+  # h.dig? :a, :b, :c # => [10, 20]
+  # h.dig? "x", "z"   # => nil
```
Comment (Sija) [inline src/named_tuple.cr:248]: ditto

```suggestion
  # h.dig? "a", "x"   # => nil
```

File: src/named_tuple.cr [hunk 4]
```diff
@@ -262,9 +262,9 @@ struct NamedTuple
   # raises `KeyError`.
   #
   # ```
-  # h = {a: {b: [10, 20, 30]}}
-  # h.dig "a", "b"                # => [10, 20, 30]
-  # h.dig "a", "b", "c", "d", "e" # raises KeyError
+  # h = {a: {b: {c: [10, 20]}}, x: {a: "b"}}
+  # h.dig :a, :b, :c # => [10, 20]
+  # h.dig "x", "z"   # raises KeyError
```
Comment (Sija) [inline src/named_tuple.cr:267]:
```suggestion
  # h.dig "a", "x"   # raises KeyError
```

File: src/time.cr (hunk 1 above)
Comment (maiha) [reply src/time.cr]: Nice catch! You're right. 👍 

This was a transitional fix before I supported new time literals and forgot to revert.
Updated(Reverted).

**Review by straight-shoota — APPROVED** [review]

**Review by HertzDevil — APPROVED** [review]

**Pull request merged**